File: front-end/src/types/AttemptResponseDTO.ts
import type { AnswerResponseDTO } from "./AnswerResponseDTO";
import type { AttemptStatus } from "./AttemptStatus";

export interface AttemptResponseDTO {
  id: number;
  exerciseId: number;
  exerciseTitle?: string;
  studentId: number;
  studentName?: string;
  attemptNumber: number;
  status: AttemptStatus;
  startedAt?: string;
  submittedAt?: string;
  gradedAt?: string;
  timeSpent?: number;
  remainingTime?: number;
  score: number;
  percentage: number;
  passed: boolean;
  teacherFeedback?: string;
  totalQuestions?: number;
  answeredQuestions?: number;
  answers?: AnswerResponseDTO[];
  createdAt: string;
}

File: back-end/src/main/java/com/meutcc/backend/content/attempt/AttemptResponseDTO.java
package com.meutcc.backend.content.attempt;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.meutcc.backend.content.answer.AnswerResponseDTO;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record AttemptResponseDTO(
        Long id,
        Long exerciseId,
        String exerciseTitle,
        Long studentId,
        String studentName,
        String studentEmail,
        Integer attemptNumber,
        String status,
        LocalDateTime startedAt,
        LocalDateTime submittedAt,
        LocalDateTime gradedAt,
        Integer timeSpent,
        Integer remainingTime,
        BigDecimal score,
        BigDecimal percentage,
        Boolean passed,
        String teacherFeedback,
        Integer totalQuestions,
        Integer answeredQuestions,
        List<AnswerResponseDTO> answers,
        LocalDateTime createdAt
) {}